[Bug] celeborn-cpp-client cannot be compiled for the ARM architecture.

### Component Selection

- [ ] Core Engine (Expression eval, Memory, Vector)
- [ ] Connectors / File Formats (Hive, Parquet, etc.)
- [ ] API / Bindings (Python, etc.)
- [x] Build
- [ ] Other

### Describe the Bug

The compilation code of the celebcpp-client does not contain the logic for determining the ARM architecture. Therefore, when `make release` is executed on the ARM architecture, `-msse=4.2` is incorrectly used, causing a compilation error.

### Error message

```
-- Build files have been written to: /root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release

celeborn-cpp-client/main-20251212: Running CMake.build()
celeborn-cpp-client/main-20251212: RUN: cmake --build "/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release" -- -j320
gmake[2]: Entering directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
gmake[3]: Entering directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
gmake[4]: Entering directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
gmake[4]: Entering directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
gmake[4]: Leaving directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
gmake[4]: Leaving directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
gmake[4]: Entering directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
gmake[4]: Entering directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
[  2%] Generating celeborn/proto/TransportMessagesCpp.pb.cc, celeborn/proto/TransportMessagesCpp.pb.h
[  5%] Building CXX object celeborn/memory/CMakeFiles/memory.dir/ByteBuffer.cpp.o
**c++: error: unrecognized command-line option ‘-msse4.2’**
gmake[4]: *** [celeborn/memory/CMakeFiles/memory.dir/build.make:79: celeborn/memory/CMakeFiles/memory.dir/ByteBuffer.cpp.o] 
gmake[4]: Leaving directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
gmake[3]: *** [CMakeFiles/Makefile2:328: celeborn/memory/CMakeFiles/memory.dir/all] Error 2
gmake[3]: *** Waiting for unfinished jobs....
gmake[4]: Leaving directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
[  5%] Built target generate_proto_source
gmake[3]: Leaving directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'
gmake[2]: *** [Makefile:101: all] Error 2
gmake[2]: Leaving directory '/root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release'


celeborn-cpp-client/main-20251212: ERROR: 
Package '727f0023958065aa68c93d41f06ba147d2abff11' build failed
celeborn-cpp-client/main-20251212: WARN: Build folder /root/.conan2/p/b/celeb70f5432e7ab46/b/build/Release
ERROR: celeborn-cpp-client/main-20251212: Error in build() method, line 81
	cmake.build()
	ConanException: Error 2 while executing
make[1]: *** [Makefile:85: build] Error 1
make[1]: Leaving directory '...../gluten'
make: *** [Makefile:106: release] Error 2
```
The location of `CXX_FLAGS`：
```
root@server1:~/.conan2/p/b/celeb1406f00bdab57/b/build/Release# grep -rl "msse4.2"
celeborn/network/CMakeFiles/network.dir/flags.make
celeborn/proto/CMakeFiles/proto.dir/flags.make
celeborn/utils/CMakeFiles/utils.dir/flags.make
celeborn/conf/CMakeFiles/conf.dir/flags.make
celeborn/protocol/CMakeFiles/protocol.dir/flags.make
celeborn/client/CMakeFiles/client.dir/flags.make
celeborn/memory/CMakeFiles/memory.dir/flags.make
```

### Reproduction Steps

1. Choose to build gluten-bolt on an ARM server, using the latest branch for the bolt.
2. Execute the gluten build commands make bolt-recipe and make release.
3. The error occurs when compiling the celeborn-cpp-client.

### Bolt Version / Commit ID

main branch @ commit 2073be2

### System Configuration

```markdown
- **OS**: (e.g. Velinux)
- **Compiler**: (GCC 12)
- **Build Type**: (Release)
- **CPU Arch**: (ARM64)
```

### Logs / Stack Trace

```shell

```

### Expected Behavior

_No response_

### Additional context

_No response_

--- a/scripts/conan/recipes/celeborn-cpp-client/all/patches/cmake.patch
+++ b/scripts/conan/recipes/celeborn-cpp-client/all/patches/cmake.patch
@@ -10,7 +10,7 @@ index f24a8abff..8a13783bf 100644
 +if (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "AMD64")
 +    message(STATUS "Detected ${CMAKE_SYSTEM_PROCESSOR} platform, enabling -msse4.2")
 +    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.2")
-+elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64" OR CMAKE_SYSTEM_PROCESSOR STREQUAL "arm64")
++elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64.*|AARCH64.*|arm64.*)")
 +    message(STATUS "Detected ${CMAKE_SYSTEM_PROCESSOR} platform, enabling -march=armv8.3-a")
 +    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=armv8.3-a")
 +else()

